### ISSUE TITLE
Is it possible to deserialize a custom class from MessagePackObject

### ISSUE THREAD

**Issue by leonidumanskiy** [msgpack/msgpack-cli#90]
Hi,
I know how to deserialize a custom class from a Stream:

var serializer = MessagePackSerializer.Get<MyClass>();
var value = serializer.Unpack(stream);

Is it possible to deserialize a class the same way directly from the MessagePackObject and not from the stream?

**Comment by yfakariya** [comment]
No, there are no `Unpacker`s to support it. But it sounds good.

**Comment by leonidumanskiy** [comment]
I am asking because I have a basic "rpc-like" messaging system, and I want to serialize an object from one of the parameters. So it looks like this:
{
"command": "create_user", 
    "parameters": {
        "hash": 1234, 
        "user": {"user_name": "test_user", "password": "1234", "email": "test@test.com"}
    }
}

So I need to receive a command first and than deserialize a User object from the second parameter.

**Comment by yfakariya** [comment]
Sorry I missed the additional question. I recommend you to implement following class:
```c#
public class Foo {
  public string Command { get; set; }
  public Dictionary<string, MessagePackObject> Parameters { get; set; }
}
```
and then, you check `Parameters` and use. I agree that it is not convinient at all, so I'll add `ObjectUnpacker` like API in future release.

**Comment by leonidumanskiy** [comment]
@yfakariya that's what I have right now.
And that's a problem I described in my first message - after taking MessagePackObject from the Parameters dictionary, I can't deserialize it into the class/object.
Instead, I have to write 
```C#
var user = new User(Parameters["user_name"].AsString(), Parameters["password"].AsString(), Parameters["email"].AsString());
```
which is really annoying.

**Comment by TSnake41** [comment]
Hi, I've found a solution if you know the type
```C#
MessagePackObject obj;
Type type;
object output;
using (MemoryStream stream = new MemoryStream())
{
     using (Packer packer = Packer.Create(stream))
     {
            obj.PackToMessage(packer, new PackingOptions());
            stream.Position = 0;
            output = SerializationContext.CreateClassicContext().GetSerializer(type).Unpack(stream);
      }
}
```

You got an object with the right type.

**Comment by yfakariya** [comment]
I will add helper extension methods in next release. It will be based on @TSnake41's idea. Thank you.

**Comment by yfakariya** [comment]
This was implemented in 0.9.
